django_freeradius/tests/test_api.py

```diff
@@ -52,3 +54,140 @@ def test_postauth_400(self):
         response = self.client.post(reverse('freeradius:postauth'), {})
         self.assertEqual(RadiusPostAuth.objects.all().count(), 0)
         self.assertEqual(response.status_code, 400)
+
+    def test_accounting_start(self):
+        self.assertEqual(RadiusAccounting.objects.all().count(), 0)
+        original = RadiusAccounting.objects.create(
+            session_id='35000006', unique_id='75058e505ac30c5b4b9c686ee853b70f',
+            username='admin', start_time='2017-08-08 15:16:10+0200',
+            nas_ip_address='172.16.64.91', nas_port_id='1',
+            nas_port_type='Async',
+            session_time='262',
+            authentication='authentication', realm='',
+            input_octets='9900909',
+            output_octets='1513075509', calling_station_id='5c:7d:c1:72:a7:3b',
+            called_station_id='00-27-22-F3-FA-F1:hostname',
+            terminate_cause='terminate_cause',
+            service_type='service_type', framed_protocol='',
+            framed_ip_address='', groupname=''
+        )
+        # we need to reload to cast values to correct python objects (see date)
+        original.refresh_from_db()
+
+        data = {'username': 'admin', 'nas_ip_address': '172.16.64.91',
+                'nas_port': '1', 'called_station_id': '00_27_22_F3_Fa_F1',
+                'calling_station_id': '5c:7d:c1:72:a7:3b', 'nas_identifier': '',
+                'acct_status_type': 'Start', 'acct_authentic': 'RADIUS',
+                'acct_delay_time': '0',
+                'unique_id': '75058e505ac30c5b4b9c686ee853b70f',
+                'acct_terminate_cause': 'User_Request', 'acct_input_octets': '1111909',
+                'acct_output_octets': '1511074444', 'nas_port_type': 'async',
+                'acct_session_time': '261', 'login_service': 'Telnet',
+                'login_ip_host': '172.16.64.25', 'session_id': '35000006',
+                'framed_protocol': '', 'framed_ip_address': '',
+                'service_type': 'Login-User', 'Realm': ''}
+
+        # sending an accounting packet
+        response = self.client.post(reverse('freeradius:accounting'), data=data)
+        self.assertEqual(response.status_code, 200)
+
+        # this makes sure that no additional objects have been created
+        self.assertEqual(RadiusAccounting.objects.filter(session_id='35000006').count(), 1)
```

Comment:
here and below we need to used ``unique_id`` field and value to retrieve the object, as it's the field we use to match existing records